WonderMake/Functionalities/Functionality.h

```diff
@@ -5,40 +5,51 @@
 class Object;
 
 //[Nicos]: Describes a functionality for an object. Template params are 1. Type that is self type and 2. Dependencies.
-template<typename TSelfType, typename ... TDependencies>
+template<typename TSelfType, typename TPolicySet>
 class Functionality
 	: public _BaseFunctionality
 {
 public:
 	inline virtual void Destroy(Object& aObject) override final;
 
-	using Super = Functionality<TSelfType, TDependencies...>;
+	using Super = Functionality<TSelfType, TPolicySet>;
 
-	template<typename TDependency>
-	constexpr __forceinline TDependency& Get() const
+	template<
+		typename TDependency,
+		typename = std::enable_if_t<
+			TPolicySet::template HasPolicy_v<TDependency, Policy::EPermission::Write>
+			|| TPolicySet::template HasPolicy_v<TDependency, Policy::EPermission::Unrestricted>>>
+	constexpr __forceinline TDependency& Get()
```

Comment:
This can be made cleaner with concepts requires syntax

```
template<typename TDependency> requires requires TPolicySet::template HasPolicy_v<TDependency, Policy::EPermission::Write> || TPolicySet::template HasPolicy_v<TDependency, Policy::EPermission::Unrestricted>>>
	constexpr __forceinline TDependency& Get()
```